static/app/components/replaysOnboarding/useLoadOnboardingDoc.tsx

```diff
@@ -0,0 +1,70 @@
+import {useEffect, useState} from 'react';
+
+import {Docs} from 'sentry/components/onboarding/gettingStartedDoc/types';
+import {Organization, PlatformIntegration, ProjectKey} from 'sentry/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
+
+function useLoadOnboardingDoc({
+  platform,
+  organization,
+  projectSlug,
+}: {
+  organization: Organization;
+  platform: PlatformIntegration;
+  projectSlug: string;
+}) {
+  const [module, setModule] = useState<null | {
+    default: Docs<any>;
+  }>(null);
+
+  const platformPath =
+    platform?.type === 'framework'
+      ? platform?.id === 'capacitor'
+        ? `capacitor/capacitor`
+        : platform?.id.replace(`${platform.language}-`, `${platform.language}/`)
+      : `${platform?.language}/${platform?.id}`;
+
+  const {
+    data: projectKeys,
+    isError: projectKeysIsError,
+    isLoading: projectKeysIsLoading,
+  } = useApiQuery<ProjectKey[]>([`/projects/${organization.slug}/${projectSlug}/keys/`], {
+    staleTime: Infinity,
+  });
```

Comment:
Nit: I have the feeling this should not be part of this hook (separation of concerns) 🤔

Reply:
yeah this API call could definitely be extracted into a hook, it's used in a lot of places!